src/util/invariant.cpp

```diff
@@ -75,32 +75,18 @@ static bool output_demangled_name(
 #endif
 
 
-/// Checks that the given invariant condition holds and prints a back trace
-/// and / or throws an exception depending on build configuration.
-/// Does not return if condition is false.
-/// Returns with no output or state change if true.
+/// Prints a back trace to 'out'
 ///
 /// \param file : C string giving the name of the file.
 /// \param function : C string giving the name of the function.
 /// \param line : The line number of the invariant
-/// \param condition : The result of evaluating the invariant condition.
-/// \param reason : C string giving the reason why the invariant should be true.
-void check_invariant(
+/// \param out: Stream to print backtrace
+void print_backtrace(
```

Comment:
This solution works on gcc and Windows: http://stacktrace.sourceforge.net/
Even if you don't use it it might be worth adding a comment mentioning it in the `#else` section below.